tests/ui/drop/drop_order.rs

```diff
@@ -1,6 +1,11 @@
 //@ run-pass
 //@ compile-flags: -Z validate-mir
+//@ revisions: edition2021 edition2024
+//@ [edition2021] edition: 2021
+//@ [edition2024] compile-flags: -Z unstable-options --cfg edition2024
+//@ [edition2024] edition: 2024
 #![feature(let_chains)]
+#![cfg_attr(edition2024, feature(if_let_rescope))]
```

Comment:
Problem [REV3,ED2024]: does not need to be `edition2024` cfg-gated if not tied to edition.